Src/DependencyCollector/Shared.Tests/Implementation/FrameworkHttpProcessingTest.cs

```diff
@@ -51,6 +51,12 @@ public void Cleanup()
 
         #region BeginEndCallBacks
 
+        [TestMethod]
+        public void OnBeginDoesNotThrowForIncorrectUrl()
+        {
+            this.httpProcessingFramework.OnBeginHttpCallback(100, "BadUrl"); // Should not throw
```

Comment:
Just curious why a bad URL should not throw? (I'm sure I'm just missing some context)
Is it because the below code in FrameworkHttpProcessing.cs is doing the try catch now?  That looks good. :)